docs/src/pages/components/data-grid/rows/ServerSortingGrid.js

```diff
@@ -69,20 +69,18 @@ export default function ServerSortingGrid() {
     };
   }, [sortModel, data]);
 
-  if (data.columns.length === 0) {
-    return null;
-  }
-
   return (
     <div style={{ height: 400, width: '100%' }}>
-      <DataGrid
-        rows={rows}
-        columns={data.columns}
-        sortingMode="server"
-        sortModel={sortModel}
-        onSortModelChange={handleSortModelChange}
-        loading={loading}
-      />
+      {data.columns.length === 0 ? null : (
```

Comment:
there should be the no rows layout

Reply:
It crashes

Reply:
It's likely a bug, I can open a new issue to keep track of it.

Reply:
FYI We have a no-rows-no-cols story in storybook `datasets`

Reply:
https://deploy-preview-338--material-ui-x.netlify.app/storybook/?path=/story/x-grid-tests-dataset--no-rows-no-cols

Reply:
Seems the problem is gone. No idea. I'm removing the defensive logic.